### PR TITLE
Stand alone Pyomo viewer

### PR DESCRIPTION
**Pull request by eslickj**
## Summary/Motivation:
Minor update to model viewer to provide an option to launch it with an embedded Jupyter qtconsole widget. To do this just run the pyomo_viewer.py script.  

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: pyomo/contrib/viewer/model_browser.py [hunk 1]
```diff
@@ -22,52 +22,7 @@
...
-        from PyQt5.QtWidgets import QAbstractItemView
-        from PyQt5.QtCore import QAbstractItemModel
-        from PyQt5 import uic
-    except:
-        _log.exception("Cannot import PyQt5")
-        class QAbstractItemModel(object):
-            def __init__(*args, **kwargs):
-                pass
-        QtCore = DummyQtCore
+from pyomo.contrib.viewer.pyqt_4or5 import *
```
Comment (jsiirola) [inline pyomo/contrib/viewer/model_browser.py]: @eslickj: the PR is missing `pyqt_4or5`.

Comment (eslickj) [reply pyomo/contrib/viewer/model_browser.py]: Oops.  Should have all files now.

File: pyomo/contrib/viewer/pyomo_viewer.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+"""
```
Comment (jsiirola) [inline pyomo/contrib/viewer/pyomo_viewer.py]: Missing copyright statement.

File: pyomo/contrib/viewer/pyqt_4or5.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
+"""
```
Comment (jsiirola) [inline pyomo/contrib/viewer/pyqt_4or5.py]: Missing copyright statement

File: pyomo/contrib/viewer/tests/test_data_model_item.py [hunk 4]
```diff
@@ -29,10 +29,46 @@ def blackbox(a,b):
...
         m.c2 = Constraint(expr=m.z[2]**4 * m.z[1]**2 + m.z[1] == 8+sqrt(2.0))
-
+        m.c3 = Constraint(expr=m.x[1] == 3)
         self.m = m.clone()
 
     def test_expr_calc(self):
         cdi = ComponentDataItem(parent=None, ui_setup=None, o=self.m.b1.e1)
         cdi.calculate()
         assert(abs(cdi.get("value")-3) < 0.0001)
+
```
Comment (jsiirola) [inline pyomo/contrib/viewer/tests/test_data_model_item.py]: We missed it the first time, but this file is also missing a copyright statement.

**Comment by eslickj** [comment]
Copyright statements have been added and updated.  I'll keep trying to improve testing in the futre.  I should be able to get a lot better coverage with a little time to write more test.  The codecove report is pretty helpful.

File: pyomo/contrib/viewer/model_browser.py [hunk 5]
```diff
@@ -295,10 +265,15 @@ def _set_ub_callback(self, val):
...
 
     def _set_active_callback(self, val):
-        if val == "True" or val == "true" or val == "1":
-            val = True
-        else:
+        if not val or val == "False" or val == "false" or val == "0" or \
+            val == "f" or val == "F":
+            # depending on the version of Qt, you may see a combo box that
+            # lets you select true/false or may be able to type the combo
+            # box will return True or Flase, if you have to type could be
```
Comment (blnicho) [inline pyomo/contrib/viewer/model_browser.py]: Typo: False

File: pyomo/contrib/viewer/model_browser.py [hunk 6]
```diff
@@ -308,10 +283,15 @@ def _set_active_callback(self, val):
...
 
     def _set_fixed_callback(self, val):
-        if val == "True" or val == "true" or val == "1":
-            val = True
-        else:
+        if not val or val == "False" or val == "false" or val == "0" or \
+            val == "f" or val == "F":
+            # depending on the version of Qt, you may see a combo box that
+            # lets you select true/false or may be able to type the combo
+            # box will return True or Flase, if you have to type could be
```
Comment (blnicho) [inline pyomo/contrib/viewer/model_browser.py]: Typo: False

File: pyomo/contrib/viewer/ui.py [hunk 7]
```diff
@@ -151,35 +132,53 @@ def __init__(self, *args, **kwargs):
...
-        self.vars = ModelBrowser(parent=self, standard="Var", ui_setup=uis)
-        self.cons = ModelBrowser(parent=self, standard="Constraint", ui_setup=uis)
-        self.params = ModelBrowser(parent=self, standard="Param", ui_setup=uis)
-        self.exprs = ModelBrowser(parent=self, standard="Expression", ui_setup=uis)
+        self.variables = ModelBrowser(parent=self, standard="Var", ui_setup=uis)
+        self.constraints = ModelBrowser(parent=self, standard="Constraint", ui_setup=uis)
+        self.parameters = ModelBrowser(parent=self, standard="Param", ui_setup=uis)
+        self.expressions = ModelBrowser(parent=self, standard="Expression", ui_setup=uis)
 
         # Dock the wigetes allong the bottom and tabify them
```
Comment (blnicho) [inline pyomo/contrib/viewer/ui.py]: Typo: widgets, along

File: pyomo/contrib/viewer/ui.py [hunk 8]
```diff
@@ -151,35 +132,53 @@ def __init__(self, *args, **kwargs):
...
+        self.addDockWidget(QtCore.Qt.BottomDockWidgetArea, self.variables)
+        self.addDockWidget(QtCore.Qt.BottomDockWidgetArea, self.constraints)
+        self.addDockWidget(QtCore.Qt.BottomDockWidgetArea, self.parameters)
+        self.addDockWidget(QtCore.Qt.BottomDockWidgetArea, self.expressions)
+        self.tabifyDockWidget(self.variables, self.parameters)
+        self.tabifyDockWidget(self.variables, self.expressions)
+        self.tabifyDockWidget(self.variables, self.constraints)
+        self.variables.raise_()
 
         # Set menu actions (rembeber the menu items are defined in the ui file)
```
Comment (blnicho) [inline pyomo/contrib/viewer/ui.py]: Typo: remember

**Pull request merged**